Classes/View/PdfView.php

```diff
@@ -127,7 +127,21 @@ private function replaceStrings($content) {
             }
         }
 
-        return $content;
+        return preg_replace_callback('/\<a (.*?)>/', function ($hit) {
+
+            $completeATag = $hit[0];
+            $returnTag = preg_replace_callback('/href="(.*)#(.*)"/', function ($href) {
+
+                if (!empty($href[2]) &&
+                        ($href[1] === '/' . str_replace('&', '&amp;', GeneralUtility::getIndpEnv(TYPO3_SITE_SCRIPT)))
+                ) {
+                    return 'href="#' . $href[2] . '"';
+                }
+            }, $completeATag);
+
+            return $returnTag;
+
+        }, $content);
```

Comment:
This solution seems very complicated to me. Could this maybe be solved in an easier way? Sth. along the following (untested):

```php
$currentUrl = preg_quote(htmlentities(GeneralUtility::getIndpEnv("TYPO3_SITE_SCRIPT")));
$currentHost = preg_quote($_SERVER['HTTP_HOST']);
$returnTag = preg_replace('/href="(?:https?:\/\/' . $currentHost . '/)?' . $currentUrl . '#(.*)"/', 'href="#\\1"');
```